test/GPv2Settlement/Settle/Reentrancy.t.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {IERC20, GPv2Order, GPv2Interaction, GPv2Settlement} from "src/contracts/GPv2Settlement.sol";
+
+import {Harness, Helper} from "../Helper.sol";
+
+import {Settlement} from "test/libraries/Settlement.sol";
+import {Sign} from "test/libraries/Sign.sol";
+import {Trade} from "test/libraries/Trade.sol";
+import {Order} from "test/libraries/Order.sol";
+
+import {SwapEncoder} from "test/libraries/encoders/SwapEncoder.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+
+// solhint-disable func-name-mixedcase
+contract Reentrancy is Helper {
+    using Settlement for Harness;
+    using Trade for GPv2Order.Data;
+    using SettlementEncoder for SettlementEncoder.State;
+    using SwapEncoder for SwapEncoder.State;
+
+    function test_revert_rejects_reentrancy_attempts_via_interactions() public {
+        reject_reentrancy_attempts_via_interactions(owner, settle_reentrancy_calldata());
+        reject_reentrancy_attempts_via_interactions(owner, swap_reentrancy_calldata());
+    }
```

Comment:
They are two different tests, so I'd execute each call in its own context to avoid that the state of the first execution pollutes the second.
Same for the function below, going from two to four tests.